### PR TITLE
Add locations, sku and remove IoT connection properties

### PR DESCRIPTION
**Pull request by visingl**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by visingl** [comment]
@dapzhang, can you please review the changes, for API readiness I have already rolled out the manifest for operation API with the new API version, for the rest, I don't know why it is red since it is a new API version, I don't expect them to fail.

**Comment by raych1** [comment]
@visingl , there's [cross version breaking change](https://github.com/Azure/azure-rest-api-specs/pull/18481#issuecomment-1083417862) and need to get breaking change review approval.

**Comment by visingl** [comment]
https://msazure.visualstudio.com/One/_workitems/edit/14027610

**Comment by visingl** [comment]
@raych1 Can you please review, its been approved by breaking change review board

**Comment by raych1** [comment]
@visingl , it's waiting for ARM review.  Please found below guide from first comment of this PR: 
"If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them."

**Comment by mentat9** [comment]
@visingl -  In future, please construct your PR for new API review by following this submission item:

**[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).**

It's fine if you don't want to use OpenAPIHub. In that case, follow this process to construct your PR:

1. Copy all files unchanged from current API version to new API version folder.
2. Make the first commit.
3. Apply changes to the files for new API version in subsequent commits, including updating the API version in each file.

This is required for ARM review of new API version PRs, but since this has already been waiting for so long, I manually set up the diffs to help unblock you sooner.

File: specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2022-04-01-preview/deviceupdate.json
Comment (mentat9) [inline specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2022-04-01-preview/deviceupdate.json:1033]: >updatePrivateEndpointProperties [](http://example.com/codeflow?start=196&length=31)

This seems better modeled by allowing update of the remotePrivateEndpoint property in the PUT (or PATCH) payload. Any particular reason you chose to model it as a separate POST action?

Comment (darkoa-msft) [reply specification/deviceupdate/resource-manager/Microsoft.DeviceUpdate/preview/2022-04-01-preview/deviceupdate.json:1033]: @mentat9 This is from Network RP specs. More specifically from a document that updated the spec without updating the spec document (if that makes any sense).  
[NRP spec (see section 7)](https://microsoft.sharepoint.com/:w:/r/teams/WAG/AzureNetworking/_layouts/15/Doc.aspx?sourcedoc=%7B64F10150-563A-4715-9476-FFA24F9BD42E%7D&file=%5BNRP%5D%20Private%20Endpoint%20Design%203.docx&action=default&mobileredirect=true&cid=7075b01e-18b3-42f7-950a-e1a09ede0e17)
[NRP move spec (see section 5)](https://microsoft.sharepoint.com/:w:/r/teams/Aznet/_layouts/15/doc2.aspx?sourcedoc=%7Bc9ed9692-4eb0-462d-a380-89375a2db68d%7D&action=view&wdAccPdf=0&wdparaid=36A48591&cid=0cea8de4-5888-4e69-af84-79c339fd99a9&share=IQGSlu3JsE4tRqOAiTdaLbaNASgh5QWY9yUgqtrJAkKJWjg&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2Cdarkoa%40microsoft.com&sourceId=&params=%7B%22AppName%22%3A%22Teams-Desktop%22%2C%22AppVersion%22%3A%2227%2F21110108719%22%7D)

I don't think any RP other than us have privateEndpointConnectionProxies API in their swagger. We have to have it because we are behind RPaaS (I can go into details, if you want).  

RPaaS does have a new feature which would allow us to remove this API from our swagger but that work is not planned yet on our side.  
RPaaS is also working on onboarding Private Endpoint Connection on behalf of all RPs.

As for the PR creation for the new version - we can do this going forward, we are planning to onboard Shift Left and, hopefully, what you are suggesting will be done automatically.

**Comment by visingl** [comment]
@raych1 let me know if everything looks good?

**Review by raych1 — APPROVED** [review]

**Pull request merged**
